array-reduce/main.js

```diff
@@ -0,0 +1,45 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const addition = (accumulator, currentValue) => accumulator + currentValue;
+const sum = numbers.reduce(addition);
+
+const multiply = (accrue, activeIndex) => accrue * activeIndex;
+const product = numbers.reduce(multiply);
+
+function transaction(accumulator, currentvalue) {
+  console.log(currentvalue.amount);
+  if (currentvalue.type === 'deposit') {
+    accumulator += currentvalue.amount;
+  } else {
+    accumulator -= currentvalue.amount;
+  }
+  return accumulator;
+}
+const balance = account.reduce(transaction, 0);
+
+function combine(accumulator, currentValue) {
+  console.log(accumulator);
+  const key = Object.keys(currentValue);
+  accumulator[key] = currentValue[key];
```

Comment:
This kind of works, but only because each of your trait objects only has one key. Take a look at `Object.assign`

```js
Object.assign(accumulator, currentValue);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign